CommandLine/XblConfig/Program.cs

```diff
@@ -0,0 +1,861 @@
+                {
+                    Console.Error.WriteLine(
+                        "Your account doesn't have access to perform the operation, please contact your administrator.");
+                }
+                else
+                {
+                    Console.WriteLine(ex.Message);
+                }
+
+                exitCode = -1;
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine("Error: unexpected error found.");
+                Console.Error.WriteLine(ex.Message);
+                exitCode = -1;
+            }
+
+            if (System.Diagnostics.Debugger.IsAttached)
+            {
+                Console.Read();
+            }
+
+            return exitCode;
+        }
+
+        private static async Task<int> GetDocumentsAsync(GetDocumentsOptions options)
+        {
+            if (options.DocumentType == DocumentType.Sandbox && string.IsNullOrEmpty(options.Sandbox))
+            {
+                throw new ArgumentException("Sandbox must be specified when obtaining sandbox documents.");
+            }
+
+            if (options.DocumentType == DocumentType.Account)
+            {
+                options.Sandbox = null;
+            }
+
+            EnsureDirectory(options.Destination);
+
+            Task<DocumentsResponse> documentsTask;
+            if (options.DocumentType == DocumentType.Sandbox)
+            {
+                Console.WriteLine("Obtaining sandbox documents.");
+                documentsTask = ConfigurationManager.GetSandboxDocumentsAsync(options.Scid, options.Sandbox);
+            }
+            else
+            {
+                Console.WriteLine("Obtaining account documents.");
+                if (options.AccountId == Guid.Empty)
+                {
+                    DevAccount user = ToolAuthentication.LoadLastSignedInUser();
+                    options.AccountId = new Guid(user.AccountId);
+                }
+
+                documentsTask = ConfigurationManager.GetAccountDocumentsAsync(options.AccountId);
+            }
+
+            using (DocumentsResponse documents = await documentsTask)
+            {
+                Console.WriteLine($"ETag: {documents.ETag}");
+                Console.WriteLine($"Version: {documents.Version}");
+                Console.WriteLine("Files: ");
+
+                foreach (var file in documents.Documents)
+                {
+                    var path = Path.Combine(options.Destination, file.Name);
+                    using (var fileStream = File.Create(path))
+                    {
+                        await file.Stream.CopyToAsync(fileStream);
+                    }
+
+                    Console.WriteLine($" - {file.Name}");
+                }
+
+                SaveETag(documents.ETag, options.Destination, options.Sandbox);
+                Console.WriteLine($"Saved {documents.Documents.Count()} files to {options.Destination}.");
+            }
+
+            return 1;

```

Comment:
>return 1; [](start = 12, length = 9)

a non 0 exit code in cmd means error.

Reply:
Oops. LOL.

---
In reply to: [175182444](https://github.com/Microsoft/xbox-live-developer-tools/pull/37#discussion_r175182444) [](ancestors = 175182444)